### PR TITLE
Some random changes to help with debugging/analysis

### PR DESCRIPTION
**Pull request by ChandraAddala**
-Fixing null pointer
-Changes for smoother graph with datastax
-Added meter to track incoming requests
-Using shaded jar for datastax
-Added some logging

### CODE REVIEW

**Comment by izrik** [comment]
Looks ok to me.

File: blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DBasicMetricsRW.java [hunk 1]
```diff
@@ -102,6 +102,9 @@ public void insertMetrics( Collection<IMetric> metrics ) throws IOException {
                 }
 
                 futures.put( locator, rawIO.insertAsync( metric ) );
+
+                // this is marking  metrics_strings & metrics_full together.
+                Instrumentation.markFullResMetricWritten();
```
Comment (shintasmith) [inline blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DBasicMetricsRW.java]: If you move this up here, is it possible we will be double-counting metrics in error cases? This means, the fullResMetricWritten count possible includes the erroneous ones?

File: blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DPreaggregatedMetricsRW.java [hunk 2]
```diff
@@ -118,7 +118,11 @@ public void insertMetrics(Collection<IMetric> metrics,
                     ResultSetFuture future = io.putAsync(locator, metric.getCollectionTime(),
                             (Rollup) metric.getMetricValue(),
                             granularity, metric.getTtlInSeconds());
+
                     futureLocatorMap.put(future, locator);
+                    if (granularity == Granularity.FULL) {
+                        Instrumentation.markFullResPreaggregatedMetricWritten();
+                    }
```
Comment (shintasmith) [inline blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DPreaggregatedMetricsRW.java]: Same here... we'll be counting the erroneous ones?

File: blueflood-http/src/main/java/com/rackspacecloud/blueflood/http/HttpResponder.java [hunk 3]
```diff
@@ -50,18 +47,19 @@ public static void respond(ChannelHandlerContext ctx, FullHttpRequest req, FullH
...
+        boolean isKeepAlive = HttpUtil.isKeepAlive(req);
+        if (isKeepAlive) {
             res.headers().add(CONNECTION, KEEP_ALIVE);
         }
 
         // Send the response and close the connection if necessary.
         ctx.channel().write(res);
-        if (req == null || !isKeepAlive(req)) {
+        if (req == null || !isKeepAlive) {
+            log.debug("Closing channel. isKeepAlive:" + isKeepAlive);
```
Comment (shintasmith) [inline blueflood-http/src/main/java/com/rackspacecloud/blueflood/http/HttpResponder.java]: Can you print the channel info here too?

File: blueflood-http/src/main/java/com/rackspacecloud/blueflood/inputs/handlers/HttpMetricsIngestionServer.java [hunk 4]
```diff
@@ -134,9 +139,10 @@ private void setupPipeline(SocketChannel channel, RouteMatcher router) {
             // if something bad happens during the decode, assume the client send bad data. return a 400.
             @Override
             public void exceptionCaught(ChannelHandlerContext ctx, Throwable thr) throws Exception {
+                log.debug("Exception during decoding", thr);
                 try {
                     if (ctx.channel().isWritable()) {
-                        log.debug("request decoder error " + thr.getCause().toString() + " on channel " + ctx.channel().toString());
+                        log.debug("request decoder error " + thr.getCause() + " on channel " + ctx.channel().toString());
```
Comment (shintasmith) [inline blueflood-http/src/main/java/com/rackspacecloud/blueflood/inputs/handlers/HttpMetricsIngestionServer.java]: I should we just print the throwable stacktrace? I.e: 
```
log.debug("....msg.....", thr);
```

File: pom.xml [hunk 5]
```diff
@@ -65,7 +65,8 @@
       <dependency>
         <groupId>com.datastax.cassandra</groupId>
         <artifactId>cassandra-driver-core</artifactId>
-        <version>3.0.3</version>
+        <version>3.1.2</version>
```
Comment (shintasmith) [inline pom.xml]: I am fine with upgrading this to 3.1.x versions, but curious, what makes you decide to go with 3.1.x, instead of staying in the 3.0.x?

**Comment by shintasmith** [comment]
Oh, also Travis seems to fail :-(. Probably just need a job restart.

File: pom.xml (hunk 5 above)
Comment (ChandraAddala) [inline pom.xml]: I just picked the latest version available.

File: blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DBasicMetricsRW.java (hunk 1 above)
Comment (ChandraAddala) [inline blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DBasicMetricsRW.java]: This is how its done with astyanax too. Thats why astyanax has smoother graphs compared to datastax. They just call this, when they add metrics to mutationBatch. Down the line mutationBatch.execute could fail.

**Comment by shintasmith** [comment]
@ChandraAddala , please also update the CHANGES.md to mention the upgrade to datastax driver 3.1.x

File: blueflood-http/src/main/java/com/rackspacecloud/blueflood/http/HttpResponder.java (hunk 3 above)
Comment (ChandraAddala) [inline blueflood-http/src/main/java/com/rackspacecloud/blueflood/http/HttpResponder.java]: Is there some specific information you have in mind? I can add the below line to the log.
```
ctx.channel().isActive()
```

File: blueflood-http/src/main/java/com/rackspacecloud/blueflood/inputs/handlers/HttpMetricsIngestionServer.java (hunk 4 above)
Comment (ChandraAddala) [inline blueflood-http/src/main/java/com/rackspacecloud/blueflood/inputs/handlers/HttpMetricsIngestionServer.java]: There is this line above the try block
```
log.debug("Exception during decoding", thr);
```

**Comment by ChandraAddala** [comment]
@shintasmith I had addressed your comments. Pls take a look

File: CHANGES.md [hunk 6]
```diff
@@ -1,6 +1,7 @@
 # CHANGES
 
 ## IN PROGRESS
+* Upgraded datastax driver to 3.1.x
```
Comment (shintasmith) [inline CHANGES.md]: We should list the exact number here.

**Pull request merged**